wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -822,6 +845,72 @@ instance ToSchema TransactionDirection where
             & type_ .~ SwaggerString
             & enum_ ?~ ["outgoing", "incoming"]
 
+-- | This is an information-less variant of 'PtxCondition'.
+data TransactionStatus
+    = Applying
+    | InNewestBlocks
+    | Persisted
+    | WontApply
+    | Creating
+    deriving (Eq, Ord, Show, Enum, Bounded)
+
+transactionStatusToText :: TransactionStatus -> Text
+transactionStatusToText x = case x of
+    Applying {} ->
+        "applying"
+    InNewestBlocks {} ->
+        "inNewestBlocks"
+    Persisted {} ->
+        "persisted"
+    WontApply {} ->
+        "wontApply"
+    Creating {} ->
+        "creating"
+
+instance ToJSON TransactionStatus where
+    toJSON x = object
+        [ "tag" .= transactionStatusToText x
+        , "data" .= Object mempty
+        ]
```

Comment:
I've used the `tag/data` representation here to allow maximum flexibility for extensions.